packages/visual-group/src/group-helper/group-utils.js

```diff
@@ -451,18 +451,26 @@ export const extractFields = (facetsAndProjections, layerFields) => {
 
 /**
  * This method sorts the facets fields inplace if field is of categorical type
- * @param {Object} facet
+ * @param {Object} facets Array of facets
  * @param {Array} keys Array of the facet field values
+ * @param {Object} config configuration object
  */
-export const sortFacetFields = (facet, keys, config) => {
-    const facetName = `${facet}`;
-    const type = facet.type();
-
-    if (type === DIMENSION && config.sort[facetName]) {
-        if (config.sort[facetName] === 'asc') {
-            keys.sort((a, b) => a - b);
-        } else {
-            keys.sort((a, b) => b - a);
-        }
-    }
+export const sortFacetFields = (facets, keys, config) => {
+    /**
+     * Check if the facet sorted by the user is plotted
+     * If an incorrect field is sorted, return the keys as is
+     */
+    const fieldsSorted = Object.keys(config.sort);
+    const allFacetsFields = facets.map(facet => `${facet}`);
+    const validFacets = fieldsSorted.filter(field => allFacetsFields.includes(field));
+    if (!validFacets.length) return;
```

Comment:
Why is this check needed?

Reply:
`validFacets` contains only the facets fields sorted by the user.
In a case where user has not sorted any facet field this might be empty.

Since `[].forEach` handles this, so this check can be removed.